src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -423,6 +423,15 @@ public void Dispose()
                 _parent._managedBuffers.Push(_buffer);
                 _buffer = null;
             }
+            
+            public void Kill()
+            {
+                if (_buffer == null || _parent.Disposed)
+                    return;
+
+                _parent?._managedBuffers?.Push(_buffer);
```

Comment:
This should not be called during the dispose, I think

Reply:
it's not a dispose call, it's a resources release code (structs don't have finalizers..)

Reply:
I'm changing it so it will just call the _buffer.Dispose, in order to prevent concurrency issues with the disposal of the context

Reply:
No need, `ManagedPinnedBuffer` already have a finalizer

Reply:
you are right